src/main/java/org/jenkinsci/remoting/nio/FifoBuffer.java

```diff
@@ -333,7 +333,7 @@ public void write(byte[] buf, int start, int len) throws InterruptedException, I
 
             synchronized (lock) {
                 while ((chunk = Math.min(len,writable()))==0)
-                    lock.wait();
+                    lock.wait(100);
```

Comment:
In line 424 there is also a <code>read()</code> call without a timeout, It also acquires the lock.